tests/deploy/e2e/test_autostarted.py

```diff
@@ -315,3 +360,745 @@ async def test_autostart_clear_agent_venv_on_delete(
+    snippetcompiler,
+    server,
+    ensure_resource_tracker_is_started,
+    client,
+    clienthelper,
+    environment,
+    auto_start_agent: bool,
+    async_finalizer,
+):
+    """
+    Verify that the new scheduler does not alter the state of agent after a restart:
+        - The agent is deploying something that takes a lot of time
+        - The agent is paused
+        - The server (and thus the scheduler) is (are) restarted
+        - The agent should remain paused (the Scheduler shouldn't do anything after the restart)
+    """
+    current_pid = os.getpid()
+
+    # First, configure everything
+    config.Config.set("config", "environment", environment)
+    # Make sure the session with the Scheduler is there
+    agentmanager = server.get_slice(SLICE_AGENT_MANAGER)
+    assert len(agentmanager.sessions) == 1
+
+    # Retrieve the actual processes before deploying anything
+    start_state = construct_scheduler_children(current_pid)
+    for children in start_state.children:
+        assert children.is_running()
+
+    snippetcompiler.setup_for_snippet(
+        """
+import minimalwaitingmodule
+
+a = minimalwaitingmodule::Sleep(name="test_sleep", agent="agent1", time_to_sleep=120)
+""",
+        autostd=True,
+    )
+
+    # Now, let's deploy a resource
+    version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    async def are_resources_deploying(deployed_resources: int = 1) -> bool:
+        result = await client.resource_list(environment, deploy_summary=True)
+        assert result.code == 200
+        summary = result.result["metadata"]["deploy_summary"]
+        deployed = summary["by_state"]["deploying"]
+        return deployed == deployed_resources
+
+    # Wait for this resource to be deployed
+    await retry_limited(are_resources_deploying, 5)
+
+    # Executors are reporting to be deploying before deploying the first executor, we need to wait for them to be sure that
+    # something is moving
+    await wait_for_consistent_children(current_pid)
+
+    async def wait_for_actual_deployment() -> bool:
+        current_children_mapping = construct_scheduler_children(current_pid)
+        return len(current_children_mapping.children) == 3
+
+    await retry_limited(wait_for_actual_deployment, 5)
+
+    # Retrieve the current processes, we should have more processes than `start_children`
+    children_after_deployment = construct_scheduler_children(current_pid)
+
+    result = await client.agent_action(tid=environment, name="agent1", action=AgentAction.pause.value)
+    assert result.code == 200
+
+    # Let's pretend that the server crashes
+    await asyncio.wait_for(server.stop(), timeout=20)
+    ibl = InmantaBootloader(configure_logging=False)
+    async_finalizer.add(partial(ibl.stop, timeout=20))
+    # Let's restart the server
+    await ibl.start()
```

Comment:
this is a clean shutdown, not a crash, but I guess getting closer to real crash is very hard. 

This is more a test for restart than crash